### ISSUE TITLE
rdkafka instrumentation 0.4.1 fails to install

### ISSUE THREAD

**Issue by arielvalentin** [open-telemetry/opentelemetry-ruby-contrib#730]
Though pushing the gem to RubyGems worked we are unable to install the gem. 

```console
vibora ~/github/opentelemetry-ruby-contrib/releases(main|✔) %
🤘 bundle
Fetching gem metadata from https://rubygems.org/.......
Could not find gem 'opentelemetry-instrumentation-rdkafka (= 0.4.1)' in rubygems repository https://rubygems.org/ or installed locally.

The source contains the following gems matching 'opentelemetry-instrumentation-rdkafka':
  * opentelemetry-instrumentation-rdkafka-0.1.0
  * opentelemetry-instrumentation-rdkafka-0.2.0
  * opentelemetry-instrumentation-rdkafka-0.2.1
  * opentelemetry-instrumentation-rdkafka-0.2.2
  * opentelemetry-instrumentation-rdkafka-0.2.3
  * opentelemetry-instrumentation-rdkafka-0.3.0
  * opentelemetry-instrumentation-rdkafka-0.3.1
  * opentelemetry-instrumentation-rdkafka-0.3.2
  * opentelemetry-instrumentation-rdkafka-0.4.0
vibora ~/github/opentelemetry-ruby-contrib/releases(main|✔) %
🤘 gem search opentelemetry-instrumentation-rdkafka

*** REMOTE GEMS ***

opentelemetry-instrumentation-rdkafka (0.4.1)
```

The Toys Release libraries prevent me from generating another release so I yanked the gem from RubyGems. 

```console
vibora ~/github/opentelemetry-ruby-contrib/instrumentation/rdkafka(main|✔) %
🤘 gem yank -v 0.4.1 opentelemetry-instrumentation-rdkafka
Yanking gem from https://rubygems.org...
Added yank_rubygem scope to the existing API key
Successfully deleted gem: opentelemetry-instrumentation-rdkafka (0.4.1)
```

RubyGems does not allow replacing or overriding a gem. The toys library does not allow me to create a new release due to failing builds, though the failing builds are excluded from the required_builds setting:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/.toys/.data/releases.yml#L9

Trying to figure out how to fix all of this. Any ideas @dazuma ?

**Comment by arielvalentin** [comment]
I wasn't able to figure out the incantations that would have resulted in the release jobs to pick up the changes and force a release of the gem. 

I created this PR and merged it:
https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/731

The release hook actions ran but because I did not set the right labels I do not think the workflows completed correctly and did not push them. I still had to push it manually.
